pod-operation/src/state_machine.rs

```diff
@@ -215,14 +217,28 @@ impl StateMachine {
 	/// Perform operations when the pod is running
 	fn _running_periodic(&mut self) -> State {
 		info!("Rolling Running state");
+
 		let encoder_value = self.wheel_encoder.measure().expect("wheel encoder faulted"); // Read the encoder value
-		if encoder_value > STOP_THRESHOLD {
+
+		// Predict next tick's braking distance
+		let current_velocity = self.wheel_encoder.get_velocity();
+		let predicted_velocity =
+			current_velocity + BRAKING_DECELERATION * TICK_INTERVAL.as_secs_f32();
+		let predicted_braking_distance = -predicted_velocity.powi(2) / (2.0 * BRAKING_DECELERATION);
+
+		// Check if the predicted braking distance requires stopping
+		if encoder_value + current_velocity * TICK_INTERVAL.as_secs_f32() >= STOP_THRESHOLD {
+			return State::Stopped;
+		}
+
+		if predicted_braking_distance <= BRAKING_THRESHOLD {
 			return State::Stopped;
 		}
```

Comment:
Suggestion: could we extract this logic to another helper method which returns a boolean indicating whether or not the pod should stop?